components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/events/TokenRevocationNotifierImpl.java

```diff
@@ -0,0 +1,156 @@
+        extractNotifiers();
+
+    }
+
+    /**
+     * Method to extract the config properties of revoke token notifiers
+     */
+    private void extractNotifiers() {
+
+        //Default Constant Values
+        final String DEFAULT_PERSISTENT_NOTIFIER_HOSTNAME = "https://localhost:2379/v2/keys/jti/";
+        final String DEFAULT_TTL = "3600";
+        final String DEFAULT_PERSISTENT_NOTIFIER_USERNAME = "root";
+        final String DEFAULT_PERSISTENT_NOTIFIER_PASSWORD = "root";
+
+        for (Map.Entry<String, Properties> entry : tokenRevocationNotifiers.entrySet()) {
+            if (entry.getKey().equalsIgnoreCase("RealtimeNotifier")) {
+                Properties properties = entry.getValue();
+                realtimeNotifierTTL = properties.getProperty("ttl", DEFAULT_TTL);
+            }
+            if (entry.getKey().equalsIgnoreCase("PersistentNotifier")) {
+                Properties properties = entry.getValue();
+                persistentNotifierHostname = properties.getProperty("hostname", DEFAULT_PERSISTENT_NOTIFIER_HOSTNAME);
+                persistentNotifierTTL = properties.getProperty("ttl", DEFAULT_TTL);
+                persistentNotifierUsername = properties.getProperty("username", DEFAULT_PERSISTENT_NOTIFIER_USERNAME);
+                persistentNotifierPassword = properties.getProperty("password", DEFAULT_PERSISTENT_NOTIFIER_PASSWORD);
+            }
+        }
+    }
+
+    /**
+     * Method to publish the revoked token on to the realtime message broker
+     *
+     * @param revokedToken requested revoked token
+     */
+    @Override public void
+    sendMessageOnRealtime(String revokedToken) {
+
+        Object[] objects = new Object[] { revokedToken, realtimeNotifierTTL };
+        Event tokenRevocationMessage = new Event(APIConstants.TOKEN_REVOCATION_STREAM_ID, System.currentTimeMillis(),
+                null, null, objects);
+        ServiceReferenceHolder.getInstance().getOutputEventAdapterService()
+                .publish(APIConstants.TOKEN_REVOCATION_EVENT_PUBLISHER, Collections.EMPTY_MAP, tokenRevocationMessage);
+        log.debug("Successfully sent the revoked token notification on realtime");
+    }
+
+    /**
+     * Method to send the revoked token to the persistent storage
+     *
+     * @param revokedToken token to be revoked
+     */
+    @Override public void
+    sendMessageToPersistentStorage(String revokedToken) {
+        String etcdEndpoint = persistentNotifierHostname + revokedToken;
+        URL etcdEndpointURL = new URL(etcdEndpoint);
+        String etcdEndpointProtocol = etcdEndpointURL.getProtocol();
+        int etcdEndpointPort = etcdEndpointURL.getPort();
+        HttpClient etcdEPClient = APIUtil.getHttpClient(etcdEndpointPort, etcdEndpointProtocol);
+        HttpPut httpETCDPut = new HttpPut(etcdEndpoint);
+        byte[] encodedAuth = Base64.encodeBase64((persistentNotifierUsername + ":" + persistentNotifierPassword).
+                getBytes(StandardCharsets.ISO_8859_1));
```

Comment:
why `ISO_8859_1` specifically?

Reply:
not to any specific reason, it was how it was done in class which i referred 
<img width="860" alt="Screen Shot 2019-04-24 at 10 00 30 AM" src="https://user-images.githubusercontent.com/23296184/56632537-e32c8d80-6677-11e9-9af1-e71299e2c3ef.png">

Reply:
Let's use UTF 8 then.

Reply:
okay, will change

Reply:
resolved